cmd/cmd_object.go

```diff
@@ -475,25 +481,58 @@ func uploadFile(bucketName, objectName, filePath, urlInfo string, ctx *cli.Conte
 	}
 	defer reader.Close()
 
-	if err = gnfdClient.PutObject(c, bucketName, objectName,
-		objectSize, reader, opt); err != nil {
-		return toCmdErr(err)
+	// if the file is more than 2G , it needs to force use resume uploading
+	if objectSize > maxPutWithoutResumeSize {
+		opt.DisableResumable = false
 	}
 
-	// Check if object is sealed
-	timeout := time.After(15 * time.Second)
-	ticker := time.NewTicker(2 * time.Second)
+	if opt.DisableResumable {
+		progressReader := &ProgressReader{
+			Reader:      reader,
+			Total:       objectSize,
+			StartTime:   time.Now(),
+			LastPrinted: time.Now(),
+		}
+
+		if objectSize > objectLargerSize {
+			progressReader.LastPrinted = time.Now().Add(2 * time.Second)
+		}
+
+		if err = gnfdClient.PutObject(c, bucketName, objectName,
+			objectSize, progressReader, opt); err != nil {
+			return toCmdErr(err)
+		}
+	} else {
+		if err = gnfdClient.PutObject(c, bucketName, objectName,
+			objectSize, reader, opt); err != nil {
+			return toCmdErr(err)
+		}
+	}
+
+	if bypassSeal {
+		fmt.Printf("\nupload %s to %s \n", objectName, urlInfo)
+		return nil
+	}
 
+	// Check if object is sealed
+	timeout := time.After(1 * time.Hour)
```

Comment:
does it need to be that long?

Reply:
After our discussion, we changed the seal printing strategy here. Users can manually cancel the wait for seal or set --bypassSeal now. If they do not cancel, they will wait until seal by default, so this time can actually be infinite.